### PR TITLE
protocol and error changes for finalized tlf support in chat CORE-3823

### PR DESCRIPTION
**Pull request by mmaxim**
@maxtaco @patrickxb r?

This adds some new stuff to the chat protocols for supporting finalized TLFs.

### CODE REVIEW

File: go/libkb/rpc_exim.go [hunk 1]
```diff
@@ -416,6 +416,21 @@ func ImportStatusAsError(s *keybase1.Status) error {
...
 			UID: uid,
 		}
+	case SCChatTLFFinalized:
+		var tlfID chat1.TLFID
+		for _, field := range s.Fields {
+			switch field.Key {
+			case "TlfID":
+				val, err := hex.DecodeString(field.Value)
+				if err != nil {
+					G.Log.Warning("error parsing chat tlf finalized TLFID")
```
Comment (patrickxb) [inline go/libkb/rpc_exim.go]: include the error in output

File: go/libkb/rpc_exim.go [hunk 2]
```diff
@@ -416,6 +416,21 @@ func ImportStatusAsError(s *keybase1.Status) error {
...
+				}
+				tlfID = chat1.TLFID(val)
```
Comment (patrickxb) [inline go/libkb/rpc_exim.go]: put this in an `else`?

File: go/libkb/errors.go [hunk 3]
```diff
@@ -1642,3 +1642,13 @@ type ChatAlreadyDeletedError struct {
...
 }
+
+//=============================================================================
+
+type ChatTLFFinalizedError struct {
+	TlfID chat1.TLFID
+}
+
+func (e ChatTLFFinalizedError) Error() string {
+	return fmt.Sprintf("unable to create conversation on finalized TLF: %s", e.TlfID.String())
```
Comment (patrickxb) [inline go/libkb/errors.go]: don't need `.String()`

File: go/libkb/rpc_exim.go (hunk 2 above)
Comment (patrickxb) [inline go/libkb/rpc_exim.go]: and should that stop the loop?  i'm not really familiar with the incoming data...

Comment (mmaxim) [inline go/libkb/rpc_exim.go]: Added the `else`, but the loop should just end after this anyway. I copied how the `GenericAPIError` case was handled.

**Pull request merged**